src/components/ceremony/ceremony.repository.ts

```diff
@@ -32,18 +31,24 @@ export class CeremonyRepository extends DtoRepository<
       actualDate: input.actualDate,
       canDelete: true,
     };
-    return await this.db
+    const result = await this.db
       .query()
       .apply(await createNode(Ceremony, { initialProps }))
-      .return<{ id: ID }>('node.id as id')
+      .return<{ node: UnsecuredDto<Ceremony> }>('node')
       .first();
+
+    if (!result) {
+      throw new ServerException('failed to create a ceremony');
+    }
+    return result;
   }
 
-  async update(
-    existing: Ceremony,
-    changes: ChangesOf<Ceremony, UpdateCeremony>,
-  ) {
-    return await this.updateProperties(existing, changes);
+  async update(changes: UpdateCeremony) {
+    const { id, ...simpleChanges } = changes;
+    await this.updateProperties({ id }, simpleChanges);
+    // I don't understand why something like funding-account does not require args but this one does?
+    // I also don't feel like I know what args or where they should come from so feel stuck
+    return await this.readOne(id);
```

Comment:
Help here would be great @CarsonF

Reply:
Ok so `readOne/Many` all call `hydrate` to populate the shape of the object that the app needs. Sometimes `hydrate` needs some more info though, like the current user to know if you are a project member or not.
That is the case here, we can see that `hydrate` arg below. 
You can see that `readOne/Many`
https://github.com/SeedCompany/cord-api-v3/blob/1e2dfedf2cc5273df3550df9bc47c1dcaf7349aa/src/core/database/dto.repository.ts#L87-L87
reference the `HydrateArgs` generic, which is defined here
https://github.com/SeedCompany/cord-api-v3/blob/1e2dfedf2cc5273df3550df9bc47c1dcaf7349aa/src/components/ceremony/ceremony.repository.ts#L24-L24

So in this case we should also pass `session` in to this `update` method, so `readOne` can use it.

`FundingAccount` may have been too simple of an example for this case.

I know there's a lot of inconsistencies between the module repositories.
But we aim to solve that with this migration.